src/H5B.c

```diff
@@ -1434,7 +1434,7 @@ H5B__remove_helper(H5F_t *f, haddr_t addr, const H5B_class_t *type, int level, u
         ret_value = H5B_INS_NOOP;
 
     /* Patch keys in neighboring trees if necessary */
-    if (*lt_key_changed && H5_addr_defined(bt->left)) {
+    if (*lt_key_changed && bt != NULL && H5_addr_defined(bt->left)) {
```

Comment:
I'm guessing GCC is warning about line 1354 above where `bt` can get set to NULL, but I guess we don't currently exercise that case in testing?

Reply:
I would assume that leads to the warning. I'm guessing that is not tested